### ISSUE TITLE
`import.meta.url` should not result in `Parsing error`

### ISSUE THREAD

**Issue by vassudanagunta** [typescript-eslint/typescript-eslint#3335]
- [ X] I have tried restarting my IDE and the issue persists.
- [ X] I have updated to the latest version of the packages.
- [ X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

<details>
<summary>Configuration</summary>

```js
{
    "root": true,
    "parser": "@typescript-eslint/parser",
    "plugins": [
        "@typescript-eslint",
        "import",
        "babel"
    ],
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended"
    ],
    "rules": {
        // adapts ESLint rules to support experimental ES features.
        // See https://github.com/babel/eslint-plugin-babel for other updated rules
        "babel/semi": [2, "never"],
        "babel/new-cap": 1,
        "babel/camelcase": 1,
        "babel/no-invalid-this": 2,
        "babel/object-curly-spacing": 1,
        "babel/no-unused-expressions": 1,
        "import/no-nodejs-modules": 2,
        "@typescript-eslint/no-explicit-any": "error"
    }
}
```

</details>

```TS
const __dirname = path.dirname(fileURLToPath(import.meta.url))
```

**Expected Result**

no error for the above line. 

**Actual Result**

```
  32:46  error  Parsing error: Unexpected token import

✖ 1 problem (1 error, 0 warnings)
```

**Additional Info**

Please see eslint/eslint/issues/14553, in particular the comments about tee

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.22.0` |
| `TypeScript`                | `X.Y.Z` |
| `ESLint`                    | `7.25.0` |
| `node`                      | `16.0.0` |

**Comment by bradzacher** [comment]
I'm unable to reproduce this against master.

> TypeScript | X.Y.Z

Likely your TS version isn't right - but it's hard to tell as you haven't filled out the issue template.

Also as mentioned in the issue template:

https://github.com/typescript-eslint/typescript-eslint/blob/1c1b572c3000d72cfe665b7afbada0ec415e7855/.github/ISSUE_TEMPLATE/typescript-eslint-parser.md#L73-L78

**Comment by vassudanagunta** [comment]
@bradzacher Brad my apologies for the delay as well as for not including a debug log in the first place. I misunderstood the the question "Did eslint throw an exception?" to be asking whether it *had* thrown an exception (I saw the error message, no exception) and that running with `--debug` was I should do if I had.

Anyway, I ran it and it does indeed throw an exception. I've update the issue description up top with the debug output, and I also update the package version info to reflect what I have currently / the environment that produced the debug dump.

Let me know what else I can do to help. I'll be far more responsive. Thanks.

**Comment by bradzacher** [comment]
```
eslint:linter SyntaxError: Unexpected token import
    at Espree.raise (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/espree/lib/espree.js:190:25)
    at Espree.unexpected (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/espree/lib/espree.js:235:18)
    at Espree.pp$3.parseExprAtom (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/acorn/dist/acorn.js:2322:21)
    at Espree.pp$3.parseExprSubscripts (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/acorn/dist/acorn.js:2129:21)
    at Espree.pp$3.parseMaybeUnary (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/acorn/dist/acorn.js:2106:19)
    at Espree.pp$3.parseExprOps (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/acorn/dist/acorn.js:2041:21)
    at Espree.pp$3.parseMaybeConditional (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/acorn/dist/acorn.js:2024:21)
    at Espree.pp$3.parseMaybeAssign (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/acorn/dist/acorn.js:1997:21)
    at Espree.pp$3.parseExprList (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/acorn/dist/acorn.js:2830:20)
    at Espree.pp$3.parseSubscript (/Users/vas/Repos/projects/TextPlainly/TextPlain.js/node_modules/acorn/dist/acorn.js:2184:27) +4ms
  eslint:linter Generating fixed text for /Users/vas/Repos/projects/TextPlainly/TextPlain.js/test/styles/test-stylesheets.js (pass 1) +1ms
```

Espree/acorn is the default parser (i.e. not our parser)
It does not support the import meta language feature.

**Comment by vassudanagunta** [comment]
🤦🏾‍♂️  @bradzacher , i am so sorry for wasting your time.   I don't know or remember how I got to this state: I have all the required pieces in my package.json, but my eslintrc is simply not configured to use `typesript-eslint`. I know I did in the beginning (as I remember seeing all the linting warnings I'm seeing (again) now.

I just reread `typesript-eslint` docs 🤦🏾‍♂️. I hope it is consolation to you that, like you, I'm working on an open source project pro-bono.
